### PR TITLE
refactor webpack configs

### PR DESCRIPTION
**Pull request by micooz**
Changes by this pull request:

* refactor webpack config into two files for different purpose.
* move `plugin` to `demo/plugin` because it's not a part of DPlayer.
* `demo` is served directly without any prefix in development.

> NOTE: paths in docs should be updated after merge this pr.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+window.DPlayer = require('./DPlayer');
```
Comment (DIYgod) [inline src/index.js]: Why use `window` directly rather than `umd`?

File: webpack/dev.config.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
+const path = require('path');
+const webpack = require('webpack');
+
+module.exports = {
+
+  devtool: 'cheap-module-source-map',
```
Comment (DIYgod) [inline webpack/dev.config.js]: It should indented with 4 spaces.

Comment (micooz) [inline webpack/dev.config.js]: now eslint well check webpack folder as well

File: src/index.js (hunk 1 above)
Comment (micooz) [inline src/index.js]: rollback to umd now

**Pull request merged**